apps/storybook/src/Table.stories.tsx

```diff
@@ -3389,3 +3414,128 @@ StickyColumns.decorators = [
   ),
 ];
+
+export const StatusAndCellIcons: Story<Partial<TableProps>> = (args) => {
+  const columns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+            cellRenderer: (
+              props: CellRendererProps<{
+                startIcon: JSX.Element;
+                endIcon: JSX.Element;
+              }>,
+            ) => (
+              <DefaultCell
+                startIcon={props.cellProps.row.original.startIcon}
+                endIcon={props.cellProps.row.original.endIcon}
+                {...props}
+              />
+            ),
+          },
+          {
+            id: 'modified',
+            Header: 'Modified',
+            accessor: 'modified',
+            maxWidth: 200,
+            cellRenderer: (
+              props: CellRendererProps<{
+                status: 'positive' | 'warning' | 'negative' | undefined;
+              }>,
+            ) => {
+              return (
+                <DefaultCell
+                  {...props}
+                  status={props.cellProps.row.original.status}
+                />
+              );
+            },
+          },
+          {
+            id: 'size',
+            Header: 'Size',
+            maxWidth: 200,
+            accessor: 'size',
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const data = useMemo(
+    () => [
+      {
+        name: 'beta.mp3',
+        modified: 'Just now',
+        size: '15 KB',
+        startIcon: <SvgSoundLoud fill='#66c6ff' />,
+      },
+      {
+        name: 'gamma.pdf',
+        modified: 'A few moments ago',
+        size: '9 MB',
+        startIcon: <SvgDetails fill='#dd3e39' />,
+        endIcon: <SvgStatusSuccess />,
+        status: 'positive',
+      },
+      {
+        name: 'delta.jpg',
+        modified: 'A few moments ago',
+        size: '963 MB',
+        startIcon: <SvgDetails fill='#7957a3' />,
+        endIcon: <SvgStatusWarning />,
+        status: 'warning',
+      },
+      {
+        name: 'Theta.dgn',
+        modified: 'A few moments ago',
+        size: '64 KB',
+        startIcon: <SvgDetails fill='#d16c00' />,
+        endIcon: <SvgStatusError />,
+        status: 'negative',
+      },
+    ],
+    [],
+  );
+
+  const rowProps = useCallback(
+    (
+      row: Row<{
+        name: string;
+        modified: string;
+        size: string;
+        startIcon: JSX.Element;
+        endIcon: JSX.Element;
+        status: string | undefined;
+      }>,
+    ) => {
+      return {
+        status: row.original.status,
+      };
+    },
+    [],
+  );
+
+  return (
+    <Table
+      {...args}
+      columns={columns}
+      data={data}
+      emptyTableContent='No data.'
+      selectionMode='multi'
+      isSelectable={true}
+      rowProps={rowProps}
+    />
+  );
+};
+
+StatusAndCellIcons.args = {
+  isSelectable: true,
+  selectionMode: 'multi',
+};
```

Comment:
That is because we are adding `filter: 'grayscale(100%)'` to svgs in disabled cells.
Maybe we need to consider using `fill` for svgs? @veekeys @mayank99